staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/versionconverter.go

```diff
@@ -41,7 +39,23 @@ func NewVersionConverter(t TypeConverter, o runtime.ObjectConvertor, h schema.Gr
 	return &versionConverter{
 		typeConverter:   t,
 		objectConvertor: o,
-		hubVersion:      h,
+		hubGetter: func(from schema.GroupVersion) schema.GroupVersion {
+			return schema.GroupVersion{
+				Group:   from.Group,
+				Version: h.Version,
+			}
+		},
+	}
+}
+
+// NewCRDVersionConverter builds a VersionConverter for CRDs from a TypeConverter and an ObjectConvertor.
+func NewCRDVersionConverter(t TypeConverter, o runtime.ObjectConvertor, h schema.GroupVersion) merge.Converter {
+	return &versionConverter{
+		typeConverter:   t,
+		objectConvertor: o,
+		hubGetter: func(from schema.GroupVersion) schema.GroupVersion {
```

Comment:
do we need this? I've asked this question already, but I'm really curious :-)

Reply:
We need this if we don't want to make two copies of the version converter that differ just by their logic around hub versions. The call to convert a crd to the hub version should be free anyway, when the hub version is the same as either the source version or the destination version.

Reply:
Also I wanted to preserve the exact behavior we have now for CRDs since this is just a bug fix and I didn't want to make it controversial. we can definitely remove this in a future release as an optimization